packages/manager/src/features/CloudPulse/Utils/utils.ts

```diff
@@ -567,6 +568,41 @@ export const filterKubernetesClusters = (
     .sort((a, b) => a.label.localeCompare(b.label));
 };
 
+/**
+ * @param serviceType The service type
+ * @param entityType The entity type
+ * @returns The filter function for the service type and entity type if applicable
+ */
+export const getFilterFn = (
+  serviceType?: CloudPulseServiceType | null,
+  entityType?: AssociatedEntityType
+) => {
+  if (!serviceType) {
+    return undefined;
+  }
+  if (serviceType === 'firewall' && entityType) {
+    return (resources: Firewall[]) =>
+      filterFirewallResources(resources, entityType);
+  }
+  if (serviceType === 'linode') {
+    return (resources: Linode[]) => filterLinodeResources(resources);
+  }
+  return undefined;
+};
+
+/**
+ * @param linodes The list of linodes
+ * @returns True if there are any linodes with aclp alerts, false otherwise
+ */
+export const filterLinodeResources = (linodes: Linode[]): Linode[] => {
+  return linodes.filter((linode) => {
+    return (
+      (linode.alerts.system_alerts?.length ?? 0) > 0 ||
+      (linode.alerts.user_alerts?.length ?? 0) > 0
```

Comment:
@ankita-akamai , will this not work

```suggestion
      linode.alerts.system_alerts?.length ||
      linode.alerts.user_alerts?.length 
```

Reply:
or you can go by 
```suggestion
      !(linode.alerts.system_alerts?.length === 0 && 
      linode.alerts.user_alerts?.length === 0)
```

Reply:
I copied this code from an existing utility in linodes and copilot is also recommending the existing one for being explicit & direct